extensions/src/objectDetection/index.ts

```diff
@@ -0,0 +1,161 @@
+import { ArgumentType, BlockType, ExtensionMenuDisplayDetails, extension, block, untilTimePassed, rgbToHex, RGBObject } from "$common";
+import { ObjectDetector as ObjectDetectorClass } from "@mediapipe/tasks-vision"
+import { initializeObjectDetector, getImageHelper } from './utils'
+
+const details: ExtensionMenuDisplayDetails = {
+  name: "Object Detection",
+  description: "Detects and identifies the object shown!",
+  iconURL: "Typescript_logo.png",
+  insetIconURL: "typescript-logo.svg"
+};
+
+export default class objectDetection extends extension(details, "video", "drawable", "addCostumes", "toggleVideoBlock", "setTransparencyBlock") {
+
+  /**
+   * The MediaPipe detector that is used to detect objects
+   */
+  detector: ObjectDetectorClass;
+
+  /**
+   * Tells whether the continuous detection should be on/off
+   */
+  continuous: boolean;
+
+  /**
+   * Dimensions of the video frame
+   */
+  DIMENSIONS = [480, 360];
+
+  /**
+   * The frequency at which the detector will generate new detections
+   */
+  processFreq: number;
+
+  /**
+   * Helper for creating ImageData objects (used to create drawables)
+   */
+  imageHelper: ReturnType<typeof getImageHelper>;
+
+  /**
+   * The list of drawables, which will all be displayed in the video frame
+   */
+  drawables: ReturnType<typeof this.createDrawable>[] = [];
+
+  /**
+   * The color of the detection box
+   */
+  color: string;
+
+  /**
+   * The thickness of the detection box
+   */
+  thickness: number;
+
+  /**
+   * Initializes the extension with standard values for the extension's class attributes.
+   */
+  async init() {
+    this.enableVideo()
+    this.continuous = false;
```

Comment:
All the values in here (like `false` `5` `'white'`, etc.) you can just set using initializers (so when you declare them above, as you do with `drawables`), like:

```ts
  color: string = 'white';
```